packages/go/analysis/ad/owns.go

```diff
@@ -0,0 +1,272 @@
+// Copyright 2024 Specter Ops, Inc.
+//
+// Licensed under the Apache License, Version 2.0
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package ad
+
+import (
+	"context"
+	"errors"
+
+	"github.com/specterops/bloodhound/analysis"
+	"github.com/specterops/bloodhound/analysis/impact"
+	"github.com/specterops/bloodhound/dawgs/cardinality"
+	"github.com/specterops/bloodhound/dawgs/graph"
+	"github.com/specterops/bloodhound/dawgs/ops"
+	"github.com/specterops/bloodhound/dawgs/query"
+	"github.com/specterops/bloodhound/graphschema/ad"
+	"github.com/specterops/bloodhound/graphschema/common"
+	"github.com/specterops/bloodhound/log"
+)
+
+func PostOwnsAndWriteOwner(ctx context.Context, db graph.Database, groupExpansions impact.PathAggregator) (*analysis.AtomicPostProcessingStats, error) {
+	operation := analysis.NewPostRelationshipOperation(ctx, db, "PostOwnsAndWriteOwner")
+
+	// Get the dSHeuristics values for all domains
+	dsHeuristicsCache, anyEnforced, err := GetDsHeuristicsCache(ctx, db)
+	if err != nil {
+		log.Errorf("failed fetching dsheuristics values for postownsandwriteowner: %w", err)
```

Comment:
Checking the called function, the only error case for this function is if we cant get the Domain nodes at all. I think we're actually safe to return an error here because its unlikely we're going to be able to process anything further if we cant fetch domain nodes at all